src/core_codemods/django_receiver_on_top.py

```diff
@@ -23,6 +23,11 @@ def leave_FunctionDef(
     ) -> Union[
         cst.BaseStatement, cst.FlattenSentinel[cst.BaseStatement], cst.RemovalSentinel
     ]:
+        if not self.filter_by_path_includes_or_excludes(
+            self.node_position(original_node)
+        ):
+            return original_node
+
```

Comment:
in attempting to add a unit test here I discovered that this include/exclude check will not work because a function def node has a different start/end numbers. Here's an example:

For code
```
        from django.dispatch import receiver

        @csrf_exempt
        @receiver(request_finished)
        def foo():
            pass
```
If a user wanted to exclude a specific line here, say they pass in to exclude ln5 which is line `def foo():` (not even the line for the decorator), when we hit  `leave_FunctionDef`, we need to match `pos.start.line == line and pos.end.line == line`. However, the position for this leave_FunctionDef node is `CodeRange(start=CodePosition(line=5, column=0), end=CodePosition(line=6, column=8))` (ends at ln6)

This is evidence that we need much more robust include/exclude behavior for these types of cases that are more subtle. I'm not quite prepared to implement this yet. @drdavella thoughts? Should I remove this include/exclude check for this (and any other case I find has this issue) for now?

Reply:
@clavedeluna yes you can remove those checks for now but please make sure to add TODOs. None of our tooling is currently taking advantage of line-based filtering as far as I'm aware so the impact should be minimal for now.